src/Resources/app/storefront/src/js/filter-category-select.plugin.js

```diff
@@ -105,6 +126,136 @@ export default class FilterCategorySelectPlugin extends FilterBasePlugin {
         this.listing.changeListing();
     }
 
+    /**
+     * @public
+     */
+
+    showHide(checkbox)
+    {
+        let inner_sub_cats = checkbox.parentNode.getElementsByClassName('sub-item');
+        let span = checkbox.parentNode.querySelector('#arrow');
+
+        if(checkbox.checked)
+        {
+            let split_cats = checkbox.value.split('_');
+
+            if(split_cats.length > 0)
+            {
+                Iterator.iterate(split_cats,(id) =>
+                {
+                    document.getElementById(id).checked = true;
+                })
+            }
+
+
+            this.toggleArrows(span,'down-arrow','up-arrow');
+            this.toggleShowHide(inner_sub_cats,'subcats-hide','subcats-show')
+
+        }
+        else
+        {
+
+            Iterator.iterate(inner_sub_cats,(sub_cat)=>
+            {
+                sub_cat.querySelector('.filter-category-select-checkbox').checked = false;
+
+            })
+        }
+
+
+    }
+
+    /**
+     * public
+     */
+
+    toggleArrows(elem,removeClass,addClass)
+    {
+        let span = elem;
+        if(span !== undefined && span !== null)
+        {
+            span.classList.remove(removeClass);
+            span.classList.add(addClass);
+        }
+    }
+
+    /**
+     * public
+     */
+    toggleByArrows(span)
+    {
+
+        let elem = span.nextElementSibling
+        let elem_array = [elem];
+
+        do {
+            var sibs = elem_array[elem_array.length -1].nextElementSibling
+            elem_array.push(sibs)
+        }
+        while( sibs !== null && sibs.getAttribute('class').indexOf('sub-item') > -1)
+
+        let classList = span.getAttribute('class');
+        let check = classList.indexOf('up-arrow') > -1;
+
+        if(check)
+        {
+            this.toggleArrows(span,'up-arrow','down-arrow')
+            this.toggleShowHide(elem_array,'subcats-show','subcats-hide');
+        }
+        else {
+            this.toggleArrows(span,'down-arrow','up-arrow')
+            this.toggleShowHide(elem_array,'subcats-hide','subcats-show');
+        }
+
+
+
+
+    }
+
+    /**
+     * public
+     */
+    _onArrowClick(arrow)
+    {
+        this.toggleByArrows(arrow)
+    }
+
+
+    /**
+     * public
+     */
+
+    toggleShowHide(elem,removeClass,showClass)
+    {
+        if(elem !== undefined && elem !== null) {
+            Iterator.iterate(elem, (subcats) => {
+                subcats.classList.remove(removeClass);
+                subcats.classList.add(showClass)
+            })
+        }
+    }
+
+    /**
+     * public
+     */
+
+    showChecked()
```

Comment:
The method names are overlapping with each other. There is a `showChecked`,  `showHide`, `toggleShowHide`, `toggleArrows`, `toggleByArrows`. Please make it a bit clearer in the name what each function does, for e.g. what is it hiding, or checked, as from the names it just seems they are doing similar things. Also add some short comment in the function to elaborate that a little bit more.